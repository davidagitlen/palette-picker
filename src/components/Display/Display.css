.Display {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: max-content;
  width: 25%;
}

.swatch {
  display: inline-block;
  height: 50px;
  width: 75px;
  margin: 10px auto;
}

.lock-image {
  display: inline-block;
  height: 40px;
  width: 40px;
  margin: 0 10px 15px 10px;
}

.lock-image:hover {
  border: none;
  cursor: pointer;
  transform: scale(1.1);
}

.hex-code {
  font-family: 'Source Code Pro', monospace;
  display: inline-block;
  font-size: 28px;
  line-height: 59px;
  padding: 0 0 0 10px;
  vertical-align: bottom;
}