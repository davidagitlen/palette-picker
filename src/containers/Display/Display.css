.Display {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: max-content;
  width: 25%;
}

.swatch {
  display: inline-block;
  height: 50px;
  width: 100px;
  margin: 10px auto;
}

.lock-image {
  display: inline-block;
  height: 40px;
  width: 40px;
  margin: 0 10px 15px 10px;
}

.hex-code {
  font-family: 'Source Code Pro', monospace;
  display: inline-block;
  font-size: 28px;
  line-height: 59px;
  padding: 0 0 0 10px;
  vertical-align: bottom;
}