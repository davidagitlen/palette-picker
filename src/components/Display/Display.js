import React from 'react';
import PropTypes from "prop-types";
import './Display.css';
import locked from '../../images/locked.svg';
import unlocked from '../../images/unlocked.svg';

const Display = ({ currentPalette, toggleLock }) => {
  const swatches = currentPalette.colors.map((color, i) => {
  const lockPath = color.locked ? locked : unlocked;
      return(
      <div key={'div' + i + Date.now()}>
        <input 
          className='lock-image'
          type='image' 
          alt='Lock/Unlock'
          key={'lock' + Date.now()}
          src={lockPath}
          onClick={() => toggleLock(i)}
          />
        <div 
          className='swatch'
          style={{backgroundColor:color[`hex_${i+1}`]}}
          key={i + Date.now()}
          >
        </div>
        <h3
          className='hex-code'
          key={'h3' + Date.now()}
        >
        {color[`hex_${i + 1}`]}
        </h3>
      </div>
      )
  })
  return (
    <section className="Display">
      <div>
        {swatches}
      </div>
    </section>
  )
}

export default Display;

Display.propTypes = {
  currentPalette: PropTypes.object.isRequired,
  toggleLock: PropTypes.func.isRequired
};