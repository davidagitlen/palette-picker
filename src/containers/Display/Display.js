import React from 'react';
import './Display.css';

const Display = () => {
  return (
    <section className="Display">
      <h2>Display</h2>
    </section>
  )
}

export default Display;