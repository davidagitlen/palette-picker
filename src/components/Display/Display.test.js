import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});

describe('Display', () => {

  let wrapper, currentPaletteOne, currentPaletteTwo, mockToggleLock;

  beforeEach(() => {
    currentPaletteOne = {
      palette: "Mock_Palette",
        project_name: "Mock_Project",
          colors: [
            { hex_1: '#ABABAB', locked: false },
            { hex_2: '#CDCDCD', locked: false },
            { hex_3: '#EFEFEF', locked: false },
            { hex_4: '#121212', locked: false },
            { hex_5: '#343434', locked: false }
          ]
    };
    currentPaletteTwo = {
      palette: "Mock_Palette",
        project_name: "Mock_Project",
          colors: [
            { hex_1: '#565656', locked: true },
            { hex_2: '#787878', locked: false },
            { hex_3: '#909090', locked: true },
            { hex_4: '#ABABAB', locked: false },
            { hex_5: '#CDCDCD', locked: true }
          ]
    };
    mockToggleLock = jest.fn();
    global.Date = {
      now: jest.fn().mockImplementation(() => 12345)
    };
    wrapper = shallow(
      <Display
        currentPalette={currentPaletteOne}
        toggleLock={mockToggleLock}
      />);
  });

  it('should match the snapshot with data passed in', () => {
    const wrapper = shallow(
      <Display
        currentPalette={currentPaletteOne}
        toggleLock={mockToggleLock}
      />); 

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if a palette with locked hexes is passed in', () => {
    const wrapper = shallow(
      <Display
        currentPalette={currentPaletteTwo}
        toggleLock={mockToggleLock}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('lock-image input', () => {
    it('should fire toggleLock on click with the index passed through as an argument', () =>{

      wrapper.find('.lock-image').first().simulate('click');

      expect(mockToggleLock).toHaveBeenCalledWith(0);
    });
  });
});