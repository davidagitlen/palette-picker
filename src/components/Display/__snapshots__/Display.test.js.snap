// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should match the snapshot if a palette with locked hexes is passed in 1`] = `
<section
  className="Display"
>
  <div>
    <div
      key="div012345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="locked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12345"
        style={
          Object {
            "backgroundColor": "#565656",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #565656
      </h3>
    </div>
    <div
      key="div112345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12346"
        style={
          Object {
            "backgroundColor": "#787878",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #787878
      </h3>
    </div>
    <div
      key="div212345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="locked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12347"
        style={
          Object {
            "backgroundColor": "#909090",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #909090
      </h3>
    </div>
    <div
      key="div312345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12348"
        style={
          Object {
            "backgroundColor": "#ABABAB",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #ABABAB
      </h3>
    </div>
    <div
      key="div412345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="locked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12349"
        style={
          Object {
            "backgroundColor": "#CDCDCD",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #CDCDCD
      </h3>
    </div>
  </div>
</section>
`;

exports[`Display should match the snapshot with data passed in 1`] = `
<section
  className="Display"
>
  <div>
    <div
      key="div012345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12345"
        style={
          Object {
            "backgroundColor": "#ABABAB",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #ABABAB
      </h3>
    </div>
    <div
      key="div112345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12346"
        style={
          Object {
            "backgroundColor": "#CDCDCD",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #CDCDCD
      </h3>
    </div>
    <div
      key="div212345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12347"
        style={
          Object {
            "backgroundColor": "#EFEFEF",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #EFEFEF
      </h3>
    </div>
    <div
      key="div312345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12348"
        style={
          Object {
            "backgroundColor": "#121212",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #121212
      </h3>
    </div>
    <div
      key="div412345"
    >
      <input
        alt="Lock/Unlock"
        className="lock-image"
        key="lock12345"
        onClick={[Function]}
        src="unlocked.svg"
        type="image"
      />
      <div
        className="swatch"
        key="12349"
        style={
          Object {
            "backgroundColor": "#343434",
          }
        }
      />
      <h3
        className="hex-code"
        key="h312345"
      >
        #343434
      </h3>
    </div>
  </div>
</section>
`;
